OneSignalSDK/onesignal/core/src/main/java/com/onesignal/internal/OneSignalImp.kt

```diff
@@ -234,21 +240,10 @@ internal class OneSignalImp : IOneSignal, IServiceProvider {
                 configModel!!.disableGMSMissingPrompt = _disableGMSMissingPrompt!!
             }
 
-            // "Inject" the services required by this main class
-            _location = services.getService()
-            _user = services.getService()
-            _session = services.getService()
-            iam = services.getService()
-            _notifications = services.getService()
-            operationRepo = services.getService()
-            propertiesModelStore = services.getService()
-            identityModelStore = services.getService()
-            subscriptionModelStore = services.getService()
-            preferencesService = services.getService()
-
-            // Instantiate and call the IStartableServices
-            startupService = services.getService()
-            startupService!!.bootstrap()
```

Comment:
I believe this was the only spot we were calling `StartupService.bootstrap()`, so we should delete the function as well.

Reply:
We can probably delete the entire StartupService class since it is no longer used. Should we also remove the StartupServiceTests as well?

Reply:
> We can probably delete the entire StartupService class since it is no longer used. 

We might want to keep it for what I noted in my other comment:
https://github.com/OneSignal/OneSignal-Android-SDK/pull/2127/files#r1651606179


> Should we also remove the StartupServiceTests as well?

For bootstrap() yes, unless you think we should move this code back there so we can keep the tests. They tests are very basic, so kind of touse up.

Reply:
I reworked StartupService so it handles calling bootstrap() and scheduleStart(). Also I added a test unit to ensure that scheduleStart does not block the main thread.